R/tmb-methods.R

```diff
@@ -427,3 +427,50 @@ h_residuals_response <- function(object) {
   assert_class(object, "mmrm_tmb")
   component(object, "y_vector") - unname(fitted(object))
 }
+
+
+#' @describeIn mmrm_tmb_methods Simulates from an mmrm model object
+#' @importFrom stats simulate
+#' @exportS3Method
+#'
+#' @param object an object of class 'mmrm'
+#' @param nsim number of response vectors to simulate. Defaults to 1.
+#' @param seed an object specifying if and how the random number generator should be initialized (‘seeded’).
+#' @param newdata data frame for which to evaluate predictions.
+#' @param ...
+#'
+#' @return A \code{data.frame} of dimension [n, m] where n is the number of observations
+#' fitted in the model, and m is the number \code{nsim} of simulated responses.
+simulate.mmrm_tmb <- function(object, nsim = 1,
+                          seed = NULL, newdata = object$data,
+                          ...){
+  if(!is.null(seed)) set.seed(seed)
```

Comment:
whether the seed should be reverted back to the original status? (otherwise it is causing some side effects with this seed argument)

Reply:
What's the side-effect being caused here?

Reply:
with seed not equal to NULL, it will change the global seed status

Reply:
do we need the `seed` argument at all? Would it not be cleaner to avoid that internal handling completely?